### PR TITLE
Setters: clear set-by if unspecified when setting a value

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: kyaml/setters2/set.go [hunk 1]
```diff
@@ -150,10 +150,8 @@ func (s SetOpenAPI) Filter(object *yaml.RNode) (*yaml.RNode, error) {
 		return nil, err
 	}
 
-	if s.SetBy != "" {
-		if err := def.PipeE(&yaml.FieldSetter{Name: "setBy", StringValue: s.SetBy}); err != nil {
-			return nil, err
-		}
+	if err := def.PipeE(&yaml.FieldSetter{Name: "setBy", StringValue: s.SetBy}); err != nil {
```
Comment (phanimarupaka) [inline kyaml/setters2/set.go]: Shouldn't this change break some existing tests ?

Comment (pwittrock) [reply kyaml/setters2/set.go]: :(

Comment (pwittrock) [reply kyaml/setters2/set.go]: It did in fact, but the tests have been removed when we disabled travis

**Pull request merged**